app/src/main/java/org/p2p/wallet/striga/user/StrigaStorage.kt

```diff
@@ -1,44 +1,58 @@
 package org.p2p.wallet.striga.user
 
-import androidx.core.content.edit
-import android.content.SharedPreferences
-import com.google.gson.Gson
 import org.p2p.core.utils.MillisSinceEpoch
-import org.p2p.core.utils.fromJsonReified
+import org.p2p.wallet.common.EncryptedSharedPreferences
+import org.p2p.wallet.common.LongPreference
+import org.p2p.wallet.common.ObjectEncryptedPreference
 import org.p2p.wallet.striga.user.model.StrigaUserStatusDetails
+import org.p2p.wallet.striga.wallet.models.StrigaFiatAccountDetails
+import org.p2p.wallet.striga.wallet.models.StrigaUserWallet
 
-private const val KEY_USER_STATUS = "KEY_USER_STATUS"
+private const val KEY_USER_STATUS =
+    "KEY_USER_STATUS"
+private const val KEY_USER_WALLET =
+    "KEY_USER_STATUS"
+private const val KEY_FIAT_ACCOUNT_DETAILS =
+    "KEY_FIAT_ACCOUNT_DETAILS"
 private const val KEY_SMS_EXCEEDED_VERIFICATION_ATTEMPTS_MILLIS =
     "KEY_SMS_EXCEEDED_VERIFICATION_ATTEMPTS_MILLIS"
 private const val KEY_SMS_EXCEEDED_RESEND_ATTEMPTS_MILLIS =
     "KEY_SMS_EXCEEDED_RESEND_ATTEMPTS_MILLIS"
 
 class StrigaStorage(
-    private val sharedPreferences: SharedPreferences,
-    private val gson: Gson
+    encryptedPrefs: EncryptedSharedPreferences,
 ) : StrigaStorageContract {
 
-    override var userStatus: StrigaUserStatusDetails?
-        get() = sharedPreferences.getString(KEY_USER_STATUS, null)?.let(gson::fromJsonReified)
-        set(value) {
-            sharedPreferences.edit {
-                if (value == null) {
-                    remove(KEY_USER_STATUS)
-                } else {
-                    putString(KEY_USER_STATUS, gson.toJson(value))
-                }
-            }
-        }
-
-    override var smsExceededVerificationAttemptsMillis: MillisSinceEpoch
-        get() = sharedPreferences.getLong(KEY_SMS_EXCEEDED_VERIFICATION_ATTEMPTS_MILLIS, 0L)
-        set(value) = sharedPreferences.edit(true) {
-            putLong(KEY_SMS_EXCEEDED_VERIFICATION_ATTEMPTS_MILLIS, value)
-        }
-
-    override var smsExceededResendAttemptsMillis: MillisSinceEpoch
-        get() = sharedPreferences.getLong(KEY_SMS_EXCEEDED_RESEND_ATTEMPTS_MILLIS, 0L)
-        set(value) = sharedPreferences.edit(true) {
-            putLong(KEY_SMS_EXCEEDED_RESEND_ATTEMPTS_MILLIS, value)
-        }
+    override var userStatus: StrigaUserStatusDetails? by ObjectEncryptedPreference(
+        preferences = encryptedPrefs,
+        key = KEY_USER_STATUS,
+        type = StrigaUserStatusDetails::class,
+        nullIfMappingFailed = true
+    )
+
+    override var userWallet: StrigaUserWallet? by ObjectEncryptedPreference(
+        preferences = encryptedPrefs,
+        key = KEY_USER_WALLET,
+        type = StrigaUserWallet::class,
+        nullIfMappingFailed = true
+    )
+
+    override var fiatAccount: StrigaFiatAccountDetails? by ObjectEncryptedPreference(
```

Comment:
the final result using new way. 
we explicitly say that we are using encrypted prefs
clear is called on authlogout and this storage has it's own prefs file